drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1158 @@
+		return -EIO;
+	}
+
+	/* REG_T_CONFIG */
+	regdata = 0;
+
+	if (drv_cfg->temperature) {
+		regdata |= TMAG5273_T_CH_EN_ENABLED;
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_T_CONFIG, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_2 %d", retval);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialize a TMAG5273 sensor
+ *
+ * @param dev handle to the device
+ *
+ * @retval 0 on success
+ * @retval -EINVAL if bus label is invalid
+ * @retval -EIO on communication errors
+ */
+static int tmag5273_init(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+	int retval;
+	uint8_t regdata;
+
+	if (!i2c_is_ready_dt(&drv_cfg->i2c)) {
+		LOG_ERR("could not get pointer to TMAG5273 I2C device");
+		return -ENODEV;
+	}
+
+	if (drv_cfg->trigger_conv_via_int) {
+		if (!gpio_is_ready_dt(&drv_cfg->int_gpio)) {
+			LOG_ERR("invalid int-gpio configuration");
+			return -ENODEV;
+		}
+
+		retval = gpio_pin_configure_dt(&drv_cfg->int_gpio, GPIO_INPUT);
+		if (retval < 0) {
+			LOG_ERR("cannot configure GPIO %d", retval);
+			return -EINVAL;
+		}
+	}
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_CONFIG_2, &regdata);
+	if (retval < 0) {
+		LOG_ERR("could not read device config 2 register %d", retval);
+		return -EIO;
+	}
+
+	LOG_DBG("operation mode: %d", (int)FIELD_GET(TMAG5273_OPERATING_MODE_MSK, regdata));
+
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_MANUFACTURER_ID_LSB, &regdata);
+	if (retval < 0) {
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_LSB) {
+		LOG_ERR("unexpected manufacturer id LSB 0x%X", regdata);
+		return -EINVAL;
+	}
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_MANUFACTURER_ID_MSB, &regdata);
+	if (retval < 0) {
+		LOG_ERR("could not read MSB of manufacturer id %d", retval);
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_MSB) {
+		LOG_ERR("unexpected manufacturer id MSB 0x%X", regdata);
+		return -EINVAL;
+	}
+
+	tmag5273_check_device_status(drv_cfg, &regdata);
+
+	retval = tmag5273_reset_device_status(dev);
+	if (retval < 0) {
+		LOG_ERR("could not reset DEVICE_STATUS register %d", retval);
+		return -EIO;
+	}
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_ID, &regdata);
+	if (retval < 0) {
+		LOG_ERR("could not read DEVICE_ID register %d", retval);
+		return -EIO;
+	}
+
+	drv_data->version = regdata & TMAG5273_VER_MSK;
+
+	/* magnetic measurement range based on version, apply correct one */
+	if (drv_cfg->meas_range == TMAG5273_DT_AXIS_RANGE_LOW) {
+		drv_data->xyz_range = tmag5273_range_low(drv_data->version);
+	} else {
+		drv_data->xyz_range = tmag5273_range_high(drv_data->version);
+	}
+
+	/* deactivate interrupts if necessary (will be overwritten if needed) */
+	regdata = TMAG5273_INT_MODE_NONE;
+
+	if (!drv_cfg->trigger_conv_via_int) {
+		regdata = TMAG5273_INT_MASK_INTB_PIN_MASKED;
+	}
```

Comment:
how about `regdata = drv_cfg->trigger_conv_via_int ? TMAG5273_INT_MODE_NONE : TMAG5273_INT_MASK_INTB_PIN_MASKED;`? then you probabaly don't need the comment anymore

Reply:
Actually you found a bug (which would happen if interrupt handling is enabled). Changed it.